modin/backends/pandas/query_compiler.py

```diff
@@ -2596,29 +2609,57 @@ def groupby_agg(
 
         as_index = groupby_kwargs.get("as_index", True)
 
-        def groupby_agg_builder(df):
-            # Set `as_index` to True to track the metadata of the grouping object
-            # It is used to make sure that between phases we are constructing the
-            # right index and placing columns in the correct order.
-            groupby_kwargs["as_index"] = True
+        if isinstance(agg_func, dict):
+
+            def groupby_agg_builder(df):
```

Comment:
Since `groupby_agg_builder`s almost duplicate each other, is it possible keep only one changing the line 2623 to `result = grouped_df.agg(agg_func) if isinstance(agg_func, dict) else agg_func(grouped_df, **agg_kwargs)`?

Reply:
or to make further code consistent:
```python
if isinstance(agg_func, dict):
    callable_fn = lambda df: df.agg(agg_func)
else:
    callable_fn = agg_func
```

Reply:
There are two arguments against lambda. 
1. One is that these functions require different set of arguments.
2. Another is that flake8 complains against lambda assignment to a variable.
So I made a change to use a condition inside of this function which black reformats into this ugly block:
```python
                    result = (
                        grouped_df.agg(agg_func)
                        if isinstance(agg_func, dict)
                        else agg_func(grouped_df, **agg_kwargs)
                    )
```

Reply:
Well, we also should do this in case of dict function, so technically this will not be a lambda assignment:
```python
wrap_udf_function(lambda df: df.agg(agg_func))
``` 
You also already uses that approach of [wrapping dict into a function and passing it to the base backend](https://github.com/gshimansky/modin/blob/a99ca6c638ee365c7b8df4013ee31752fef0f657/modin/backends/pandas/query_compiler.py#L2584-L2601), and that works well, why we can't do the same in pandas qc?

Reply:
@dchigarev `flake8` complains when you declare a variable like this `myfunc = lambda whatever`. Technically `wrap_udf_function` is not needed there but if I remove it I get this:
```
modin/backends/pandas/query_compiler.py:2585:17: E731 do not assign a lambda expression, use a def
                callable_func = lambda df, *args, **kwargs: df.aggregate(agg_func, *agg_args, **agg_kwargs)
                ^
```
The same happens if I write your suggestion `callable_fn = lambda df: df.agg(agg_func)`.

Reply:
we can probably mark that line as `# noqa`

Reply:
Yes we can, but I currently chose to write @YarShev suggestion with conditional operator instead of lambda and ignoring lint error. However if you insist on lambda, I can implement it.